### PR TITLE
Make CallInfo an interface, and generate fields on demand.

### PR DESCRIPTION
**Pull request by guoliang100**
@alainjobart

### CODE REVIEW

File: go/vt/callinfo/callinfo.go [hunk 1]
```diff
@@ -9,32 +9,32 @@ import (
...
-	HTML template.HTML
+	HTML() template.HTML
 }
 
 // internal type and value
-type key int
+type key string
 
-var callInfoKey key = 0
+var callInfoKey key = "vt.CallInfo"
```
Comment (alainjobart) [inline go/vt/callinfo/callinfo.go]: Why this change? The key is an interface, so it has both type and value, so just int with value 0 is enough to avoid collisions...

**Comment by alainjobart** [comment]
LGTM after comment is addressed.

File: go/vt/callinfo/callinfo.go (hunk 1 above)
Comment (guoliang100) [inline go/vt/callinfo/callinfo.go]: "0" is good enough currently. Though I think it might have collision in future.
I reverted to 0 for now.

Comment (alainjobart) [inline go/vt/callinfo/callinfo.go]: You won't get any collisions, even if someone else uses '0' as the value, as both the value AND the type need to match, and the type is private in this package, so no other package can use it...
